db/migrate/20150701173332_extend_vmt_to_tenant_relationship.rb

```diff
@@ -0,0 +1,33 @@
+class ExtendVmtToTenantRelationship < ActiveRecord::Migration
+  def up
+    # Spawn linking table
+    create_table :atmosphere_virtual_machine_template_tenants do |t|
+      t.references :virtual_machine_template
+      t.references :tenant
+    end
+    # Rewrite existing data
+    existing_templates = execute("SELECT id, tenant_id FROM atmosphere_virtual_machine_templates")
+    existing_templates.each do |t|
+      puts "Processing VMT #{t['id'].to_s} which belongs to tenant #{t['tenant_id'].to_s}"
+      execute("INSERT INTO atmosphere_virtual_machine_template_tenants(virtual_machine_template_id, \
+        tenant_id) VALUES(#{t['id'].to_s}, #{t['tenant_id'].to_s})")
+    end
+    # Get rid of old FKey
+    execute("ALTER TABLE atmosphere_virtual_machine_templates DROP COLUMN tenant_id")
+
+  end
+
+  def down
+    # Recreate tenant_id in vmts
+    add_column :atmosphere_virtual_machine_templates, :tenant_id, :integer
+    # Populate field
+    ts = execute("SELECT virtual_machine_template_id AS vmid, MIN(tenant_id) AS tid FROM \
```

Comment:
Just a remark (I'm not forcing to change it into such construction). You can use something like:

```ruby
<<-SQL
        atmosphere_appliances.id NOT IN (
          SELECT DISTINCT(appliance_id)
            FROM atmosphere_dev_mode_property_sets)
SQL
```